scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala

```diff
@@ -31,8 +31,8 @@ trait GrantHandler {
       }.getOrElse {
         handler.createAccessToken(authInfo)
       }
-      case Some(token) => Future.successful(token)
-      case None => handler.createAccessToken(authInfo)
+      case Some(token) if !token.isExpired => Future.successful(token)
```

Comment:
I'm not following why this is needed - what was the scenario that this was broken for?